### PR TITLE
Initial modification of how configs are handled for stonesense

### PR DESCRIPTION
**Pull request by realSquidCoder**
Stonesense configs are now in the dfhack-configs folder when installed but the actual files are here in the stonesense repo

### CODE REVIEW

**Comment by realSquidCoder** [comment]
dont merge til i add changelog stuff btw, unless i should still make a seperate pr (doubtful)

File: docs/changelog.txt [hunk 1]
```diff
@@ -49,3 +49,5 @@ Template for new versions:
 
 ## Misc Improvements
 - `stonesense`: changed announcements to be right-aligned and limited it to only show the most recent 10 announcements.
+- `stonesense`: config files are now in `configs/stonesense`in the stonesense repo and are in `dfhack-configs/stonesense` when installed.
```
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply docs/changelog.txt:52]: oops i forgot it was rst technically

File: main.cpp [hunk 2]
```diff
@@ -422,7 +422,7 @@ static void * stonesense_thread(ALLEGRO_THREAD * main_thread, void * parms)
     ssState.Size.x = DEFAULT_SIZE;
     ssState.Size.y = DEFAULT_SIZE;
     ssState.Size.z = DEFAULT_SIZE_Z;
-    ssConfig.show_creature_names = true;
+    //ssConfig.show_creature_names = false;
```
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply main.cpp:425]: neither are needed. its in `stoneinit.txt` keeping it in code for reference right now

File: Config.cpp [hunk 3]
```diff
@@ -801,7 +801,7 @@ void parseConfigLine( string line )
 bool loadConfigFile()
 {
     string line;
-    ALLEGRO_PATH * p =al_create_path("stonesense/init.txt");
+    ALLEGRO_PATH * p =al_create_path("dfhack-config/stonesense/stoneinit.txt");
```
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply Config.cpp:804]: it is redundant but im still deciding if i will be keeping the folder or not. keeping it stoneinit right now means i dont have to change as much later

File: main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply main.cpp]: ok point taken and change made.

File: CMakeLists.txt [hunk 4]
```diff
@@ -3,7 +3,8 @@ PROJECT (stonesense)
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake/Modules)
 
 include_directories (${CMAKE_SOURCE_DIR})
-
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/plugins/stonesense/configs/
+        DESTINATION "${DFHACK_DATA_DESTINATION}/data/dfhack-config-defaults")
```
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply CMakeLists.txt:7]: what does your change do?

[the remark this answers is not in the document]
Comment (realSquidCoder) [reply CMakeLists.txt:7]: the fact that i grab everything in the configs folder is intentional. it is a folder i created too

**Comment by realSquidCoder** [comment]
Is there any more we need to do on this or can we get it merged? I'm hoping to start work on my larger config changes soon

File: Config.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (ab9rf) [reply Config.cpp:804]: i agree with the subfolder, and i would just call it `init.txt`

File: Config.cpp [hunk 5]
```diff
@@ -801,7 +801,7 @@ void parseConfigLine( string line )
...
     const char * path = al_path_cstr(p,ALLEGRO_NATIVE_PATH_SEP);
     ifstream myfile (path);
     if (myfile.is_open() == false) {
```
Comment (ab9rf) [inline Config.cpp:807]: will this fall back to the default config in `dfhack-config-defaults/stonesense`?

File: CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply CMakeLists.txt]: a misclick which i am resolving now

File: Config.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (realSquidCoder) [reply Config.cpp:804]: same. i'll rename to init as well.

**Comment by realSquidCoder** [comment]
should i update the docs or make a fresh PR for that?

File: Config.cpp (hunk 5 above)
Comment (ab9rf) [reply Config.cpp:807]: oh right forgot about that

**Comment by realSquidCoder** [comment]
ok _now_ im actually finished. ready to be merged if no other issues are found

File: main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (myk002) [reply main.cpp]: this is going to be especially important as new options are added, but `init.txt` files don't get updated. We don't have a mechanism for "diffing" in new defaults. Once a file exists in `dfhack-config`, it won't get overwritten, even if we change the default content.

**Comment by myk002** [comment]
> Is there any more we need to do on this or can we get it merged? I'm hoping to start work on my larger config changes soon

Remember, with `git`, you're never blocked. You can branch off of this PR branch and work on a dependent feature there. If more changes get made to this PR, the updates can be merged into the other branch.

We try to get to PRs expediently, but our life schedules don't always allow it. Small PRs can go through quickly, but larger or contentious PRs can take much longer. Patience is appreciated.

File: configs/stonesense/init.txt
Comment (myk002) [inline configs/stonesense/init.txt:1]: I think Lethosor mentioned this -- the file would make more sense as `configs/init.txt` -- there's no need to duplicate the `stonesense` path in this repo. As written, I think the file will end up as `data/dfhack-config-defaults/stonesense/stonesense/init.txt`

File: docs/changelog.txt [hunk 6]
```diff
@@ -49,3 +49,5 @@ Template for new versions:
 
 ## Misc Improvements
 - `stonesense`: changed announcements to be right-aligned and limited it to only show the most recent 10 announcements.
+- `stonesense`: config files are now in ``configs/stonesense``in the stonesense repo and are in ``dfhack-configs/stonesense`` when installed.
```
Comment (myk002) [inline docs/changelog.txt:52]:
```suggestion
- `stonesense`: `init.txt` config file is now read from ``dfhack-configs/stonesense/init.txt``
```

Comment (myk002) [reply docs/changelog.txt:52]: non-public-facing changes don't need to go in the changelog (unless they have serious consequences for external developers)

File: main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ab9rf) [reply main.cpp]: The way DF handles this in most cases is to read the default config first, to set defaults, then read the player's config. This way, settings in the player's config will override the defaults, and any settings not mentioned in the player config will default to those in the default.

[the remark this answers is not in the document]
Comment (realSquidCoder) [reply main.cpp]: ok yall are raising some good points here and for this reason im converting this back to a draft as i have a newer idea i wanna look into

File: docs/changelog.txt [hunk 7]
```diff
@@ -49,3 +49,5 @@ Template for new versions:
 
 ## Misc Improvements
 - `stonesense`: changed announcements to be right-aligned and limited it to only show the most recent 10 announcements.
+- `stonesense`: `init.txt` config file is now read from ``dfhack-configs/stonesense/init.txt``
```
Comment (myk002) [inline docs/changelog.txt:52]: ha, whoops.
```
``init.txt``
```

File: configs/stonesense/init.txt
Comment (myk002) [reply configs/stonesense/init.txt:1]: bump

**Review by myk002 — COMMENTED** [review]
comment on init.txt path is not addressed yet

File: configs/stonesense/init.txt
Comment (realSquidCoder) [reply configs/stonesense/init.txt:1]: nope i changed it awhile back and it does `...defaults/stonesense/init.txt`

Comment (realSquidCoder) [reply configs/stonesense/init.txt:1]: image
ignore config.json im testing a sub branch of this one

Comment (realSquidCoder) [reply configs/stonesense/init.txt:1]: actually wait thy're both outdated

Comment (realSquidCoder) [reply configs/stonesense/init.txt:1]: i'm missing a commit, hang on thx for catching that

**Comment by realSquidCoder** [comment]
should be fixed? hopefully?

File: docs/changelog.txt [hunk 8]
```diff
@@ -49,3 +49,5 @@ Template for new versions:
 
 ## Misc Improvements
 - `stonesense`: changed announcements to be right-aligned and limited it to only show the most recent 10 announcements.
+- `stonesense`: ``init.txt`` config file is now read from ``dfhack-configs/stonesense/init.txt``
+- `stonesense`: creature names are now hidden by default and can be shown by pressing ``n`` with default keybinds.
```
Comment (myk002) [inline docs/changelog.txt:53]:
```suggestion
- `stonesense`: creature names are now hidden by default (they can still be shown by pressing ``n`` (default keybinding) in stonesense).
```

**Review by myk002 — APPROVED** [review]

**Pull request merged**